### ISSUE TITLE
Speed up CI builds

### ISSUE THREAD

**Issue by JPercival** [cqframework/cqf-ruler#464]
Currently, full CI builds take 20-25 minutes to run. We should look at ways to parallelize or otherwise speed the builds up while still maintaining test coverage.

Some ideas:
* Run static analysis separately
* Run more integration tests in parallel
* Multi-stage builds where PRs only run unit tests while potential releases run integration tests.

**Comment by JPercival** [comment]
Retiring cqf-ruler in favor of HAPI, won't fix.
